lib/pbench/agent/tool_data_sink.py

```diff
@@ -1683,89 +2059,44 @@ def main(argv):
-        redis_server.connection_pool.disconnect()
-        redis_server = None
 
     optional_md = params["optional_md"]
 
-    # Before we daemonize, flush any data written to stdout or stderr.
-    sys.stderr.flush()
-    sys.stdout.flush()
-
-    pidfile_name = f"{PROG}.pid"
-    pfctx = pidfile.PIDFile(pidfile_name)
-    with open(f"{PROG}.out", "w") as sofp, open(
-        f"{PROG}.err", "w"
-    ) as sefp, daemon.DaemonContext(
-        stdout=sofp,
-        stderr=sefp,
-        working_directory=os.getcwd(),
-        umask=0o022,
-        pidfile=pfctx,
-        files_preserve=[fh.stream.fileno()],
-    ):
-        try:
-            # We have to re-open the connection to the redis server now that we
-            # are "daemonized".
-            logger.debug("constructing Redis() object")
-            try:
-                redis_server = redis.Redis(host=redis_host, port=redis_port, db=0)
-            except Exception as e:
-                logger.error(
-                    "Unable to connect to redis server, %s:%s: %s",
-                    redis_host,
-                    redis_port,
-                    e,
-                )
-                return 8
-            else:
-                logger.debug("constructed Redis() object")
-
-            with ToolDataSink(
-                pbench_bin,
-                hostname,
-                bind_hostname,
-                redis_server,
-                redis_host,
-                redis_port,
-                channel_prefix,
-                benchmark_run_dir,
-                tool_group,
-                tool_trigger,
-                tools,
-                tool_metadata,
-                optional_md,
-                logger,
-            ) as tds_app:
-                tds_app.execute()
-        except OSError as exc:
-            if exc.errno == errno.EADDRINUSE:
-                logger.error(
-                    "ERROR - tool data sink failed to start, %s:%s already in use",
-                    bind_hostname,
-                    tds_port,
-                )
-            else:
-                logger.exception("ERROR - failed to start the tool data sink")
-        except Exception:
-            logger.exception("ERROR - failed to start the tool data sink")
-
-    return 0
+    if daemonize == "yes":
+        ret_val = daemon(
+            PROG,
+            redis_server,
+            redis_host,
+            redis_port,
+            pbench_bin,
+            pbench_run,
+            hostname,
+            tar_path,
+            cp_path,
+            param_key,
+            params,
+            optional_md,
+        )
+    else:
+        ret_val = driver(
+            PROG,
+            redis_server,
+            redis_host,
+            redis_port,
+            pbench_bin,
+            pbench_run,
+            hostname,
+            tar_path,
+            cp_path,
+            param_key,
+            params,
+            optional_md,
+        )
```

Comment:
I'm not exactly sure why, but these giant parameter lists really bug me, and having them appear twice in a row makes it worse.  How about this, which gets rid of one of them?

```
    start_func = daemon if daemonize else driver
    ret_val = start_func(
            PROG,
            redis_server,
            redis_host,
            redis_port,
            pbench_bin,
            pbench_run,
            hostname,
            tar_path,
            cp_path,
            param_key,
            params,
            optional_md,
        )
```